### PR TITLE
Update the Stripe Connect flow for GitHub Sponsored organizations

### PR DESCRIPTION
**Pull request by matchai**
Howdy Open Collective folks! 👋

Going through the flow just now, it looks like the document was out-of-date, with many fields no longer present in the Stripe Connect setup flow.

I've gone ahead and updated it, but my screenshots don't have the nice looking drop-shadows that the previous ones did. Feel free to alter the images as you see fit.

### CODE REVIEW

**Comment by znarf** [comment]
@contraexemplo could you review that?

File: collectives/github-sponsored-organizations.md [hunk 1]
```diff
@@ -32,36 +32,15 @@ This next section is asking questions about the business. As a Collective fiscal
 
 * Legal business name: `Open Source Collective`
 * Doing business as: `Name of your collective`
-* Business address:
```
Comment (contraexemplo) [inline collectives/github-sponsored-organizations.md:35]: @matchai Hi! Sorry for the delay. I have a question: I see you've deleted the lines with the business address, but the screenshot you provided shows the form still has that field. Shouldn't it stay?

Comment (matchai) [reply collectives/github-sponsored-organizations.md:35]: The form no longer asks for the business address, but instead, just the personal address of the person registering.

Comment (contraexemplo) [reply collectives/github-sponsored-organizations.md:35]: Got it. Thanks!

**Comment by contraexemplo** [comment]
Thank you @matchai for your contribution! We appreaciate it a lot. :heart:

**Pull request merged**